### DISCUSSION TITLE
Different extra for different engines?

### DISCUSSION THREAD

**Discussion by ghost** [SeaQL/sea-query#406] [Q&A] [answered]
Can I have a different `.extra("anything I want to add".to_string())` for each supported engine or do I have to decide which engine I'll be using if I use `.extra()`?

**Comment by ikrivosheev** [comment, marked as answer]
> Can I have a different `.extra("anything I want to add".to_string())` for each supported engine or do I have to decide which engine I'll be using if I use `.extra()`?

Hello! you need to decide engine before using extra, because `extra` pass to query "as is", without manipulation in sea-query.

**Comment by billy1624** [comment]
Hey @JediMaster25, for MySQL, I think you can customize the engine using this method.
https://docs.rs/sea-query/0.26.2/sea_query/table/struct.TableCreateStatement.html#method.engine
